xformers/components/feedforward/mlp.py

```diff
@@ -33,14 +40,36 @@ def __init__(
         **kwargs,
     ):
         super().__init__()
-
-        self.mlp = nn.Sequential(
-            nn.Linear(dim_model, hidden_layer_multiplier * dim_model, bias=bias),
-            build_activation(activation),
-            nn.Dropout(dropout),
-            nn.Linear(hidden_layer_multiplier * dim_model, dim_model, bias=bias),
-            nn.Dropout(dropout),
-        )
+        dim_mlp = hidden_layer_multiplier * dim_model
+        # check if fused Bias Activation Dropout is applicable
+        if xformers._is_functorch_available:
+            self.requires_cuda = True
+            self.mlp = nn.Sequential(
+                nn.Linear(
+                    in_features=dim_model, out_features=dim_mlp, bias=False
+                ),  # bias is handled in the next layer
+                NVFusedBiasActivationDropout(
+                    p=dropout,
+                    bias_shape=dim_mlp if bias else None,
+                    activation=activation,
+                ),
```

Comment:
The fact that the bias live in a different module means that you won't be able to easily swap to use nvfuser on a pre-trained model which used the standard layers. This is IMO a drawback which would be good to be solved.

Using this optimization means that this is not just a transparent change to the user, and the snippet like I proposed before with `state_dict` won't work out of the box.